src/test/java/org/humancellatlas/ingest/schemas/SchemaTest.java

```diff
@@ -0,0 +1,77 @@
+package org.humancellatlas.ingest.schemas;
+
+import com.github.tomakehurst.wiremock.junit.WireMockRule;
+import org.humancellatlas.ingest.schemas.schemascraper.SchemaScraper;
+import org.humancellatlas.ingest.schemas.schemascraper.impl.SchemaScraperImpl;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import java.io.File;
+import java.net.URI;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.Collection;
+
+import static com.github.tomakehurst.wiremock.client.WireMock.*;
+
+
+/**
+ * Created by rolando on 19/04/2018.
+ */
+@RunWith(SpringRunner.class)
+@SpringBootTest
+public class SchemaTest {
+    @Autowired SchemaService schemaService;
+
+    @Rule
+    public WireMockRule wireMockRule = new WireMockRule(8088);
+
```

Comment:
As previously suggested, use `MockRestServiceServer` instest of `WireMock`.

Reply:
I prefer WireMock. MockRestServiceServer assumes we're using RestTemplate for all HTTP calls. In the state tracker we use RestTemplate as well as Traverson for traversing HAL REST APIs, requiring a more generic HTTP mock setup.

Reply:
We're using `RestTemplate` anyway. It's something we already need to configure separately as it needs to be shared among all Spring beans that need it. The tests will have to be made aware of that implementation detail as it's part of object construction phase, regardless of how that happens. We're also using `SpringRunner` to run the test to begin with, so the code is already coupled to the framework. Unless we foresee the need to use a different client library in the near future, there seems to be little advantage to the WireMock approach. And even then, both WireMock and MockRestServiceServer seem to follow the same idiom, so porting the code over if absolutely necessary does not seem to take that much effort.

Adding Wiremock as additional dependency adds to our technical debt as it's one more external library we have to manage. Pretty much everything WireMock does, RestClientTest/MockRestServiceServer already does, with additional bonus of out of the box configuration by the framework.

Reply:
I believe we should use 1 method of mocking HTTP servers across our java projects for consistency. WireMock allows us to use Traverson, RestTemplate, and any other HTTP client. The state tracker uses Traverson and RestTemplate

Reply:
You can inject `RestTemplate` into `Traverson` using [`setRestOperations`](https://docs.spring.io/spring-hateoas/docs/current/api/org/springframework/hateoas/client/Traverson.html#setRestOperations-org.springframework.web.client.RestOperations-).